File: JavaScript/lengthOfLongestSubstring .js
 s = "pwwkeww"


 var lengthOfLongestSubstring = function(s) {

    let arr = []
    let count = 0

        for (let x of s) {
            
            if (arr.includes(x)) {
                arr.splice(0, arr.indexOf(x) + 1);
            }
            arr.push(x);
            count = Math.max(count, arr.length); 
            console.log(arr)
        }
        
        return count
    
 };

 console.log(lengthOfLongestSubstring(s))

File: Python/lengthOfLongestSubstring.py


s = "pwwkew"

def lengthOfLongestSubstring(s):

    arr = []
    count = 0

    for x in s:
        if x in arr:
            arr = arr[arr.index(x) + 1:]
        arr.append(x)        
        count = max(count, len(arr))
    return count

print(lengthOfLongestSubstring(s))    